docs/specs/src/hwp_2_overview.md

```diff
@@ -0,0 +1,284 @@
+
+System Description
+------------------
+
+We can describe the system as Agents, who play Games together by taking
+Actions, the Records of which are held in a distributed Ledger that is
+built by sharing these Records over a Network with other Agents. We
+capitalize terms that comprise the ontological units of the system, and
+which are formally described in the later sections.
+
+### Agents
+
+Agents have these properties:
+
+1.  Agents are the only source of Actions in the system, thus Agents are
+  the source of agency. All such Actions are uniquely identifiable
+  as to which Agent took them, i.e. all Actions are
+  cryptographically signed (see Actions below).
+
+2.  Agents are uniquely addressable by other Agents.
+
+3.  An Agent's address is its public key.
+
+4.  Agents share Records of the Actions they take with other Agents for
+  distributed storage so that those Records can be retrieved by
+  other Agents reliably.
+
+5.  Agents validate received Actions before storing them.
+
+6.  Agents respond to requests for stored information.
+
+7.  Agents can send messages with arbitrary content directly to other
+  Agents.
+
+### Games
+
+Games have these properties:
+
+1.  A Game consists of an Integrity specification with these parts:
+
+    a.  A deterministic description of the structure of any data that is
+  used to record some "play" in the game. Such data is called an
+  Entry, where the act of generating such data is called an
+  Action which is also recorded. Note: both types of data,
+  content of the play (Entry) and the meta-data about the play
+  (Action) when taken together are called a Record.
+
+    b.  A deterministic description of the validity of the contents of
+  an Entry along with the validity of an Agent taking any
+  Action, i.e. some Actions may to be taken in some contexts but
+  not in others, as in making a move out-of-turn, or by an Agent
+  not allowed to play a certain role.
+
+    c.  A deterministic description of how Records may be related or
+  linked to other Records.
+
+2.  Along with the Integrity specification, a Game also consists of a
+  Coordination specification. This specification contains
+  instruction sets that wrap Actions into function call units and
+  thus serve as an API to the Game. For example, for a blogging
+  "Game" one such function call might be "create\_post" which takes
+  a number of Actions that atomically create a number of Records to
+  the Agent's Source Chain which include an Entry for the post as
+  well as links relating the post to other Entries (see below for
+  definitions of Actions and Source Chain).
+
+3.  Each instance of the Game is played on it's own Game Board which
+  exists as a unique and independent network of Agents playing that
+  Game, i.e. Games cannot interact with each other directly as all
+  action in the system is only taken by Agents. Note that Games can
+  be composed together, but only by Agents playing across multiple
```

Comment:
I don't feel like this needs any extra explanation, but if there were more, maybe it could be stated that the "new game" which is created by the composition of two existing games only exists for the agents who are playing both games. Like, the game of making puns in German by inserting English words that sound like German words can only be played by people who speak both languages.

Reply:
What this is intended to point to set-up for is UIs that are assemblages of multiple DNAs.  I think I'll add a couple more sentences to make this clearer.